### PR TITLE
Update processor to run on NanoAOD with globalParT stage2

### PR DESCRIPTION
**Pull request by cmantill**
- [x] Update processor to add variables
- [x] Add filelist
- [x] Add comments to cross section list with checks from xsecdb
- [x] Enable BDT inference in processor (to speed up later in signal region processing)
- [x] Add notebook and script to compare Xbb tagger performance
- [x] Add branches with gen level information for V+Jets and diboson (not in the latest jobs yet)

### CODE REVIEW

File: src/HH4b/xsecs.py [hunk 1]
```diff
@@ -163,114 +185,61 @@
...
-
-xsecs["VBFHHto4B_CV-m2p12_C2V-3p87_C3-m5p96_TuneCP5_13p6TeV_madgraph-pythia8"] = (
-    0.056818 * BR_HBB * BR_HBB
-)
-xsecs["VBFHHto4B_CV_m2p12_C2V_3p87_C3_m5p96_TuneCP5_13p6TeV_madgraph-pythia8"] = (
-    0.056818 * BR_HBB * BR_HBB
-)
-
+# c3 is kl
+vbfhh = {
```
Comment (jmduarte) [inline src/HH4b/xsecs.py:189]: should we take xsdb numbers for everthing?

File: src/HH4b/processors/GenSelection.py [hunk 2]
```diff
@@ -64,6 +65,7 @@ def gen_selection_HHbbbb(
     fatjets: FatJetArray,
     selection_args: list,
     skim_vars: dict,
+    fatjet_str: str = "bbFatJet",
```
Comment (jmduarte) [inline src/HH4b/processors/GenSelection.py:68]: so the options are `bbFatJet` or `ak8FatJet` maybe we can make that explicit somewhere?

File: src/HH4b/run_utils.py [hunk 3]
```diff
@@ -188,6 +187,13 @@ def parse_common_args(parser):
         default="2022",
         choices=["2018", "2022", "2022EE", "2023", "2023BPix"],
     )
+    parser.add_argument(
+        "--pnet-txbb",
```
Comment (jmduarte) [inline src/HH4b/run_utils.py:191]: should we rename this argument since the options are not just pnet anymore

File: src/HH4b/run_utils.py [hunk 4]
```diff
@@ -188,6 +187,13 @@ def parse_common_args(parser):
...
+        type=str,
+        required=True,
+        choices=["legacy", "v12", "part"],
```
Comment (jmduarte) [inline src/HH4b/run_utils.py:194]: similarly, maybe the options should be `pnet-legacy`, `pnet-v12`, `glopart-v1` ? but i understand if that causes other headaches in the code

File: src/HH4b/xsecs.py (hunk 1 above)
Comment (cmantill) [reply src/HH4b/xsecs.py:189]: ok, but given that I submitted jobs with these numbers, I propose to open an issue and then solve it (taking xsdb numbers) with a separate PR

File: src/HH4b/run_utils.py (hunk 3 above)
Comment (cmantill) [reply src/HH4b/run_utils.py:191]: done

File: src/HH4b/processors/corrections.py [hunk 5]
```diff
@@ -412,38 +338,34 @@ def get_jmsr(
     fatjets: FatJetArray,
     num_jets: int,
     year: str,
```
Comment (jmduarte) [inline src/HH4b/processors/corrections.py:340]: can add a `noqa: ARG001` to pass pre-commit for unused `year` argument

**Review by jmduarte — CHANGES_REQUESTED** [review]

**Review by jmduarte — APPROVED** [review]

**Pull request merged**